### ISSUE TITLE
Language server crashing upon loading a project

### ISSUE THREAD

**Issue by saleyn** [elixir-lsp/elixir-ls#908]
### Environment

* Elixir & Erlang versions (elixir --version): 1.15.0-rc.2
* Elixir Language Server version: ElixirLS release v0.14.6
* Operating system: Arch Linux 5.15.90.1-microsoft-standard-WSL2  (running in WSL2 on Windows)
* Editor or IDE name (e.g. Emacs/VSCode): VSCode

### Current behavior

```
Starting ElixirLS release v0.14.6
Running in /home/serge/projects/elixir-libs/typed_struct
[Warn  - 9:51:18 AM] ** (MatchError) no match of right hand side value: {:error, :already_present}
    (language_server 0.14.6) lib/language_server/cli.ex:20: ElixirLS.LanguageServer.CLI.main/0
    (elixir 1.15.0-rc.2) lib/code.ex:1432: Code.require_file/2
[Error - 9:51:18 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
```

Downloaded a project https://github.com/ejpcmac/typed_struct, and tried to open it in VSCode.  The issue occurs with every Elixir project.

**Comment by lukaszsamson** [comment]
Elixir 1.15 is not yet supported.

**Comment by saleyn** [comment]
When running Elixir 1.14.5 compiled with OTP-26, I am getting this error:
```
Starting ElixirLS release v0.14.6
Running in /home/serge/projects/elixir-libs/typed_struct
[Info  - 8:45:24 PM] Started ElixirLS v0.14.6
[Info  - 8:45:24 PM] Running in /home/serge/projects/elixir-libs/typed_struct
[Info  - 8:45:24 PM] ElixirLS built with elixir "1.14.5" on OTP "26"
[Info  - 8:45:24 PM] Running on elixir "1.14.5 (compiled with Erlang/OTP 26)" on OTP "26"
[Info  - 8:45:24 PM] Protocols are not consolidated
[Warn  - 8:45:24 PM] OTP compiled without EEP48 documentation chunks. Language features for erlang modules will run in limited mode. Please reinstall or rebuild OTP with appropriate flags.
[Info  - 8:45:24 PM] Elixir sources not found (checking in /home/build/elixir). Code navigation to Elixir modules disabled.
Runtime terminating during boot ({badarg,[{io,put_chars,[standard_error,[_]],[{_},{_},{_}]},{Elixir.Kernel.CLI,exec_fun,2,[{_},{_}]},{Elixir.Kernel.CLI,run,1,[{_},{_}]},{init,start_em,1,[]},{init,do_boot,3,[]}]})

Crash dump is being written to: erl_crash.dump...done
[Error - 8:45:26 PM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
```

**Comment by lukaszsamson** [comment]
Please see compatibility matrix. OTP 26 is not supported

**Comment by saleyn** [comment]
Sorry, in the matrix it says that it should work with Elixir 1.15 and Erlang 25, but I am getting the same error as in the beginning of this issue with those versions:
```
$ asdf current erlang
erlang          25.3.2.2
$ asdf current elixir
elixir          1.15.0-rc.2-otp-25
```
Output:
```
Starting ElixirLS release v0.14.6
Running in /home/serge/projects/elixir-libs/typed_struct
[Warn  - 7:37:46 AM] ** (MatchError) no match of right hand side value: {:error, :already_present}
    (language_server 0.14.6) lib/language_server/cli.ex:20: ElixirLS.LanguageServer.CLI.main/0
    (elixir 1.15.0-rc.2) lib/code.ex:1432: Code.require_file/2
[Error - 7:37:46 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Info  - 7:37:46 AM] Connection to server got closed. Server will restart.
true
[Error - 7:37:46 AM] ElixirLS - typed_struct client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
```

However it does with with elixir "1.14.5" on OTP "25.3".  So if Elixir 1.15 and OTP 25 is not yet supported, the matrix needs to be updated to reflect that.

**Comment by lukaszsamson** [comment]
Master branch is working with 1.15 but there were no release yet

**Comment by Hermanverschooten** [comment]
Bit sad you closed this issue, as Elixir 1.15 has been released and more people will run into this issue now.
I did and had to search before coming up on this closed issue.

**Comment by take** [comment]
looks like this is fixed due to the following release

https://github.com/elixir-lsp/elixir-ls/releases/tag/v0.15.0
